lib/registration.pm

```diff
@@ -82,6 +83,19 @@ sub accept_addons_license {
     }
 }
 
+=head2 add_suseconnect_product
+
+    add_suseconnect_product($name, [$version, [$arch, [$params]]]);
+
+Wrapper for SUSEConnect -p $name.
+=cut
+sub add_suseconnect_product {
+    my ($name, $version, $arch, $params) = @_;
+    $version //= sle_version_convert();
+    $arch //= get_required_var('ARCH');
+    assert_script_run("SUSEConnect -p $name/$version/$arch $params");
```

Comment:
I like idea! As the only usage of $param in the code is regcode, I guess we can move that hard-coded part here as well. I also wonder if we can detect zypper conflicts here as we do in command_register subroutine in lib/suseconnect_register.pm. But we can leave it for future improvements.

Reply:
@rwx788 Good point, thanks. Actually I had `$regcode` param instead of generic `$params`. Then I decided to use `$params` then as there are other params used in the code. I left this one, which I want to convert (unless anyone objects that we don't need that much generic function)
`assert_script_run "SUSEConnect --de-register --product $module/$version/$arch";`

Reply:
Maybe it would be better for `add_suseconnect_product` to have only two params: `$name` and `%args`. If `version` and `arch` aren't in keys, they would get specified.
Current version actually forces to specify `$version` and `$arch` if I want to add additional params.

Reply:
And maybe using `$params` can be considered insecure (inserting pipes with commands, ... Proper way would be really having hash `%params` and use only these which are really supported by `SUSEConnect`).